consensus/istanbul/backend/announce.go

```diff
@@ -842,37 +918,37 @@ func (sb *Backend) handleSignedAnnounceVersionsMsg(peer consensus.Peer, payload
 		validAddresses[signedAnnVersion.Address] = true
 		validEntries = append(validEntries, signedAnnVersion.Entry())
 	}
-	return sb.upsertSignedAnnounceVersionEntries(validEntries)
+	if err := sb.upsertAndGossipSignedAnnounceVersionEntries(validEntries); err != nil {
+		logger.Warn("Error upserting and gossiping entries", "err", err)
+		return err
+	}
+	// If this node is a validator (checked later as a result of this call) and it receives a signed announce
+	// version from a remote validator that is newer than the remote validator's
+	// version in the val enode table, this node did not receive a direct announce
+	// and needs to announce its own enode to the remote validator.
+	sb.startGossipAnnounceTask()
+	return nil
 }
 
-func (sb *Backend) upsertSignedAnnounceVersionEntries(entries []*vet.SignedAnnounceVersionEntry) error {
+func (sb *Backend) upsertAndGossipSignedAnnounceVersionEntries(entries []*vet.SignedAnnounceVersionEntry) error {
 	logger := sb.logger.New("func", "upsertSignedAnnounceVersions")
 	newEntries, err := sb.signedAnnounceVersionTable.Upsert(entries)
 	if err != nil {
 		logger.Warn("Error in upserting entries", "err", err)
 	}
 
-	// If this node is a validator (checked later as a result of this call) and it receives a signed announce
-	// version from a remote validator that is newer than the remote validator's
-	// version in the val enode table, this node did not receive a direct announce
-	// and needs to announce its own enode to the remote validator.
-	sb.startGossipAnnounceTask()
-
-	// Only regossip entries that are new to this node and do not originate
-	// from an address that we have gossiped a signed announce version for
-	// within the last 5 minutes
+	// Only regossip entries that do not originate from an address that we have
+	// gossiped a signed announce version for within the last 5 minutes, excluding
+	// our own address.
 	var signedAnnVersionsToRegossip []*signedAnnounceVersion
 	sb.lastSignedAnnounceVersionsGossipedMu.Lock()
 	for _, entry := range newEntries {
 		lastGossipTime, ok := sb.lastSignedAnnounceVersionsGossiped[entry.Address]
-		if !ok || time.Since(lastGossipTime) >= 5*time.Minute {
-			signedAnnVersionsToRegossip = append(signedAnnVersionsToRegossip, &signedAnnounceVersion{
-				Address: entry.Address,
-				Version: entry.Version,
-				Signature: entry.Signature,
-			})
-			sb.lastSignedAnnounceVersionsGossiped[entry.Address] = time.Now()
+		if ok && time.Since(lastGossipTime) >= signedAnnounceVersionGossipCooldownDuration && entry.Address != sb.ValidatorAddress() {
```

Comment:
Should the inequality be inverted?
```suggestion
		if ok && time.Since(lastGossipTime) < signedAnnounceVersionGossipCooldownDuration && entry.Address != sb.ValidatorAddress() {
```